test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.js

```diff
@@ -0,0 +1,142 @@
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptionsForType2Transactions,
+        smartContract,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await unlockWallet(driver);
+
+        await openDapp(driver, contractAddress);
+
+        await initiateContractInteractionTx(driver);
+      },
+    );
+  });
+
+  it(`Opens a contract interaction type 2 transaction that includes layer 1 fees breakdown on a layer 2`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder({ inputChainId: CHAIN_IDS.OPTIMISM })
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .withTransactionControllerOPLayer2Transaction()
+          .build(),
+        ganacheOptions: {
+          ...defaultGanacheOptionsForType2Transactions,
+          network_id: hexToNumber(CHAIN_IDS.OPTIMISM),
+          chainId: hexToNumber(CHAIN_IDS.OPTIMISM),
+        },
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        await openAdvancedDetailsAndCheckTheyExist(driver);
+      },
+    );
+  });
+});
+
+async function deployContract(driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement(`#deployButton`);
+
+  await driver.delay(2000);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+  await driver.waitForSelector({
+    css: '.confirm-page-container-summary__action__name',
+    text: 'Contract deployment',
+  });
+
+  await driver.clickElement({ text: 'Confirm', tag: 'button' });
+  await driver.waitUntilXWindowHandles(2);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.ExtensionInFullScreenView);
+  await driver.clickElement({ text: 'Activity', tag: 'button' });
+  await driver.waitForSelector(
+    '.transaction-list__completed-transactions .activity-list-item:nth-of-type(1)',
+  );
+}
+
+async function initiateContractInteractionTx(driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement(`#depositButton`);
+
+  await driver.delay(2000);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+  await driver.waitForSelector({
+    css: 'h2',
+    text: 'Transaction request',
+  });
+}
+
+async function openAdvancedDetailsAndCheckTheyExist(driver) {
+  await driver.waitForSelector({ css: 'p', text: 'Fee details' });
```

Comment:
we could remove the wait since on the `clickElement` we have a guard where we wait until element is visible and clickable

```
const element = await this.findClickableElement(rawLocator);
....
async findClickableElement(rawLocator) {
    const element = await this.findElement(rawLocator);
    await Promise.all([
      this.driver.wait(until.elementIsVisible(element), this.timeout),
      this.driver.wait(until.elementIsEnabled(element), this.timeout),
    ]);
    return wrapElementWithAPI(element, this);
  }

```